src/Workspaces/Core/Portable/Storage/SQLite/v2/Interop/SqlConnection.cs

```diff
@@ -302,7 +303,19 @@ public Optional<Checksum.HashData> ReadChecksum_MustRunInTransaction(Database da
                 });
         }
 
-        private Stream ReadBlob(SafeSqliteBlobHandle blob)
+        private Stream ReadAndUncompressBlob(SafeSqliteBlobHandle blob)
+        {
+            using var blobStream = ReadBlobWorker(blob);
```

Comment:
Is this similar to https://phiresky.github.io/blog/2022/sqlite-zstd/ ?

That would still allow querying compressed blobs (which I don't have enough context on the impl to figure that out)

Reply:
Yes.  Except their approach sits below the application layer (you're unaware that compression is happening), while mine is at the application layer (roslyn is compressing/decompressing the blobs that are stored).